### ISSUE TITLE
Complete UNet Implementation

### ISSUE THREAD

**Issue by DanielScottEaton** [DanielScottEaton/TrenchRipper#1]

**Comment by DanielScottEaton** [comment]
UNet implementation is still a bit shaky. The loss function needs to be modified and the envelope trick (from JB) tried.

**Comment by DanielScottEaton** [comment]
Added weightmap generation and its associated hyperparameters to the grid search. Continuing to develop the grid search functionality.
